ntbs-service/wwwroot/source/Components/ValidateMultiple.ts

```diff
@@ -7,38 +7,56 @@ const qs = require("qs");
 const ValidateMultiple = Vue.extend({
     props: {
         model: String,
-        properties: Array
+        properties: Array,
+        isDateValidation: Boolean
     },
     data: function () {
         return {
-            hasError: false
+            hasError: false,
+            errorFields: []
         }
     },
     methods: {
         // Annotate values to compare with ref="input[i]", and error fields with ref="errorField[i]", for any number of inputs that should be validated together
         // Each error message will be mapped to the corresponding field.
         validate: function () {
             const inputs = this.createArrayFromRefElements("input");
-            var inputValues: any;
-            if (inputs[0].type === "radio") {
+            var queryString: any;
+            if (this.isDateValidation) {
+                var dayInputs = inputs.map((i: any) => i.$refs.dayInput.value);
+                if (arrayContainsEmptyValues(dayInputs)) {
+                    return;
+                }
+                var monthInputs = inputs.map((i: any) => i.$refs.monthInput.value);
+                if (arrayContainsEmptyValues(monthInputs)) {
+                    return;
+                }
+                var yearInputs = inputs.map((i: any) => i.$refs.yearInput.value);
+                if (arrayContainsEmptyValues(yearInputs)) {
+                    return;
+                }
+                queryString = buildKeyDateValuesQueryString(this.properties, dayInputs, monthInputs, yearInputs);
+                this.errorFields = inputs.map((i: any) => i.$refs.errorField);
+            }
+            else if (inputs[0].type === "radio") {
                 // TODO: Do this mapping for other types if element is reused for non-radio inputs.

```

Comment:
Is this comment still relevant?

Reply:
Yes because we could e.g. use it for a normal input field or a checkbox (rather than a radio button or a date field)